codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoder.java

```diff
@@ -91,17 +87,16 @@ protected void maxHeaderSizeExceeded() throws Http2Exception {
 
     @Override
     public Http2Headers decodeHeaders(ByteBuf headerBlock) throws Http2Exception {
-        InputStream in = new ByteBufInputStream(headerBlock);
         try {
             final Http2Headers headers = new DefaultHttp2Headers(validateHeaders, (int) headerArraySizeAccumulator);
             HeaderListener listener = new HeaderListener() {
                 @Override
-                public void addHeader(byte[] key, byte[] value, boolean sensitive) {
-                    headers.add(new AsciiString(key, false), new AsciiString(value, false));
+                public void addHeader(CharSequence key, CharSequence value, boolean sensitive) {
```

Comment:
or wait I will do a followup where I merge the hpack2.internal package into http2 package.